daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -540,23 +544,55 @@ func tealCompileTest(t *testing.T, bytesToUse []byte, expectedCode int, enableDe
 	req := httptest.NewRequest(http.MethodPost, "/", bytes.NewReader(bytesToUse))
 	rec := httptest.NewRecorder()
 	c := e.NewContext(req, rec)
-	err := handler.TealCompile(c)
+	err := handler.TealCompile(c, params)
 	require.NoError(t, err)
 	require.Equal(t, expectedCode, rec.Code)
+
+	// Check compiled response.
+	if rec.Code == 200 {
+		data := rec.Body.Bytes()
+		err = protocol.DecodeJSON(data, &response)
+		require.NoError(t, err, string(data))
+		if params.Sourcemap != nil && *params.Sourcemap {
```

Comment:
nit: I would make the condition `expectedSourcemap != nil`, that way you aren't replicating the same logic that determines when a sourcemap should be returned or not

Also, `require.Equal(t, expectedSourcemap, response.Sourcemap)` compares `*logic.SourceMap`, does this pointer comparison compare the actual values?

Reply:
Good catch thanks!